src/ga4gh/vrsatile/pydantic/vrsatile_models.py

```diff
@@ -222,5 +224,55 @@ class Config:
         validator('gene_context', allow_reuse=True)(return_value)
     _get_vrs_allele_ref_seq_val = \
         validator('vrs_ref_allele_seq', allow_reuse=True)(return_value)
-    _get_allelic_statee_val = \
+    _get_allelic_state_val = \
         validator('allelic_state', allow_reuse=True)(return_value)
+
+
+class CategoricalVariationType(str, Enum):
+    """Possible types for Categorical Variations."""
+
+    CANONICAL_VARIATION = "CanonicalVariation"
+    COMPLEX_VARIATION = "ComplexVariation"
+
+
+class CategoricalVariation(BaseModel):
+    """A representation of a categorically-defined `functional domain
+    <https://en.wikipedia.org/wiki/Domain_of_a_function>`_ for variation, in
+    which individual variation instances may be members.
+    """
+
+    id: Optional[CURIE] = Field(..., alias="_id")
+    type: CategoricalVariationType
+    complement: bool
+
+    _get_id_val = validator("id", allow_reuse=True)(return_value)
+
+
+class CanonicalVariation(CategoricalVariation):
+    """A categorical variation domain characterized by a representative
+    Variation context to which members lift-over, project, translate, or
+    otherwise directly align.
+    """
+
+    type: Literal[CategoricalVariationType.CANONICAL_VARIATION] = \
+        CategoricalVariationType.CANONICAL_VARIATION
+    variation: Optional[Union[Allele, Haplotype, CopyNumber,
+                              Text, VariationSet]]
+
+
+class ComplexVariationOperator(str, Enum):
+    """Possible values for the Complex Variation's `operator` field."""
+
+    AND = "AND"
+    OR = "OR"
+
+
+class ComplexVariation(CategoricalVariation):
+    """A categorical variation domain jointly characterized by two or more
+    other categorical variation domains.
+    """
+
+    type: Literal[CategoricalVariationType.COMPLEX_VARIATION] = \
+        CategoricalVariationType.COMPLEX_VARIATION
+    operands: List[CategoricalVariation]
```

Comment:
We should validate that there are at least 2 CategoricalVariation's in this field + add tests